core/src/convert.rs

```diff
@@ -1,6 +1,11 @@
 //! Conversion to and from ipld.
-use crate::cid::Cid;
+//!
+//! Please note conversion use the ```as``` keyword under the hood.
```

Comment:
Error handling would be tricky as the error type for ```ipld.try_into()``` and ```TryFrom<Ipld>``` would have to be convertible into each other. Possible to do surely but not without changes to the error handling. Maybe create a new issue?

Reply:
Hmm. Thinking from the users perspective. I probably wouldn't read the module level documentation and would expect the `TryFrom` doing what I'd expect it to do: converting things losslessly if there is no error. I feat that using `as` would lead to hard to find bugs for people using the library. E.g. "why is this integer value 524 suddenly 12". That would happen if it was converted to an u8.

Reply:
True!

I'm used to ```this_error``` for error handling IDK how to do it with anyhow. I'll look into it.